src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -423,5 +436,9 @@ private Token GetNextToken(out string tokenString)
             tokenString = sb.ToString();
             return Token.String;
         }
+
+        [DoesNotReturn]
+        private void ThrowInvalidAssemblyName()
+            => throw new FileLoadException(SR.InvalidAssemblyName, _inputAsString ?? _input.ToString());
```

Comment:
Is `?? _input.ToString()` reachable?

Reply:
It is reachable when the parser is initialized by ReadOnlySpan

Reply:
Ah, correct. `_inputAsString` will be null on this path: https://github.com/dotnet/runtime/blob/2b3e202db5318e985b114c88567c13135ae23a0b/src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs#L81